include/grpc/grpc.h

```diff
@@ -533,8 +533,27 @@ GRPCAPI char* grpc_channelz_get_subchannel(intptr_t subchannel_id);
    is allocated and must be freed by the application. */
 GRPCAPI char* grpc_channelz_get_socket(intptr_t socket_id);
 
+/************* Shutdown support. *************/
+/* Register a function to be called at the end of the last grpc_shutdown() call. */
+GRPCAPI void grpc_on_shutdown_callback(void (*func)());
+/* Run an arbitrary function on an arg */
+GRPCAPI void grpc_on_shutdown_callback_with_arg(void (*f)(const void*), const void* arg);
+
+#ifdef __cplusplus
+}
+#endif
+
 #ifdef __cplusplus
+/* should probably be somewere else (new header in src\core\lib\gprpp ??) */
```

Comment:
Yes, definitely should be somewhere else, the core surface is supposed to be pure C89.